### PR TITLE
No longer use fileinput when generating COMInterfaces as it breaks parallel builds

### LINKED ISSUE #13290: Generation of comInterfaces sometimes fails when it is being done in parallel.
Regression from PR #13226 cc @leonardder 
### Steps to reproduce:
Build NVDA from a clean checkout on a multicore machine. If the issue does not show up repeat as many times as needed.
### Actual behavior:
Occasionally build fails with the following:
```
scons: *** [source\comInterfaces\AcrobatAccessLib.py] RuntimeError : input() already active
Traceback (most recent call last):
  File "C:\Users\my_user_name\nvda\.venv\lib\site-packages\SCons\Action.py", line 1279, in execute
    result = self.execfunction(target=target, source=rsources, env=env)
  File "C:\Users\my_user_name\nvda\source\comInterfaces_sconscript", line 70, in interfaceAction
    makeIDEFriendly(path)
  File "C:\Users\my_user_name\nvda\source\comInterfaces_sconscript", line 48, in makeIDEFriendly
    for line in fileinput.input(path, inplace=True):
  File "C:\Python37\lib\fileinput.py", line 93, in input
    raise RuntimeError("input() already active")
RuntimeError: input() already active
ia2_p.c
scons: building terminated because of errors.
```

### Expected behavior:
Build should be successful every time.
#### NVDA installed/portable/running from source:
From source
#### NVDA version:
Latest master as of January 29-th
#### Windows version:
N/A
#### Name and version of other software in use when reproducing the issue:
N/A
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No such issues with versions before introduction of #13226 
#### If NVDA add-ons are disabled, is your problem still occurring?
N/A
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
N/A

## Technical:
Since `fileinput` redirects stdout to a file we cannot do this in multiple threads at the same time. I see two ways of fixing this:
- No longer use `fileinput` - read the file, create the IDE friendly version in memory and overwrite the file instead.
- Tell SCons not to generate COMInterfaces in parallel.

### PR DESCRIPTION
**Pull request by lukaszgo1**
### Link to issue number:
Fixes #13290
Related to [this thread on nvda-devel](https://groups.io/g/nvda-devel/topic/89208225#46086) by @MarcoZehe 
### Summary of the issue:
PR #13226 started building nVDA in parallel. While this offers significant speed improvements the way in which we were generating COMInterfaces was not thread safe. The issue was caused by the `fileinput` module which redirects stdout to a given file and this cannot be done in multiple threads at the same time. Also, as discovered by @MarcoZehe in the thread linked above when stdout is redirected to a file some other build step which is executing in a separate thread can write to it and the text which should be shown on screen ends up in the COMInterface file.
### Description of how this pull request fixes the issue:
Were not using `fileinput` anymore - the IDE friendly content is generated in memory and then written to the file afterwards.
### Testing strategy:
Made sure that COMInterfaces can be generated from a clean checkout.
### Known issues with pull request:
None known
### Change log entries:
None needed.
### Code Review Checklist:

- [X] Pull Request description:
  - description is up to date
  - change log entries
- [X] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [X] API is compatible with existing add-ons.
- [X] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [X] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

**Comment by seanbudd** [comment]
We are also considering reverting #13226

**Comment by lukaszgo1** [comment]
Note that even if #13226 gets reverted this PR should still be merged since, as @michaelDCurran  said, allowing people to build in parallel is still beneficial.

File: source/comInterfaces_sconscript [hunk 1]
```diff
@@ -45,15 +44,18 @@ except ModuleNotFoundError:
...
+	with open(path, "r") as interfaceFile:
+		for line in interfaceFile:
+			match = importPattern.match(line)
+			if match:
+				libId = match.group(1)
+				IDEFriendlyContent.append(importTemplate.format(libIdentifier=libId))
+			else:
+				IDEFriendlyContent.append(line)
+	with open(path, "w") as interfaceFile:
+		interfaceFile.write("".join(IDEFriendlyContent))
```
Comment (leonardder) [inline source/comInterfaces_sconscript:57]:
```suggestion
		interfaceFile.writelines(IDEFriendlyContent)
```

**Review by LeonarddeR — CHANGES_REQUESTED** [review]
Thanks for taking this!

**Comment by lukaszgo1** [comment]
@leonardder  All addressed.

**Review by LeonarddeR — APPROVED** [review]

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**